pkg/driver/node/mounter/pod_mounter.go

```diff
@@ -69,36 +73,96 @@ func NewPodMounter(podWatcher *watcher.Watcher, credProvider *credentialprovider
 //  6. Wait until Mountpoint successfully mounts at `target`
 //
 // If Mountpoint is already mounted at `target`, it will return early at step 2 to ensure credentials are up-to-date.
-func (pm *PodMounter) Mount(ctx context.Context, bucketName string, target string, credentialCtx credentialprovider.ProvideContext, args mountpoint.Args) error {
+func (pm *PodMounter) Mount(ctx context.Context, bucketName string, target string, credentialCtx credentialprovider.ProvideContext, args mountpoint.Args, fsGroup string, pvMountOptions string) error {
 	volumeName, err := pm.volumeNameFromTargetPath(target)
 	if err != nil {
 		return fmt.Errorf("Failed to extract volume name from %q: %w", target, err)
 	}
 
-	podID := credentialCtx.PodID
-
 	err = pm.verifyOrSetupMountTarget(target)
 	if err != nil {
 		return fmt.Errorf("Failed to verify target path can be used as a mount point %q: %w", target, err)
 	}
 
-	isMountPoint, err := pm.IsMountPoint(target)
+	isTargetMountPoint, err := pm.IsMountPoint(target)
 	if err != nil {
 		return fmt.Errorf("Could not check if %q is already a mount point: %w", target, err)
 	}
 
+	if isTargetMountPoint {
+		klog.V(4).Infof("Target path %q is already mounted. Only refreshing credentials.", target)
+		source, err := pm.findSourceMountPoint(target)
+		if err != nil {
+			klog.Errorf("Failed to find source mount point for %q: %v", target, err)
+			return fmt.Errorf("Failed to find source mount point for %q: %w", target, err)
+		}
+		mpPodUID := filepath.Base(source)
+		podPath := pm.podPath(mpPodUID)
+
+		mpPodLock := getMPPodLock(mpPodUID)
+		mpPodLock.mutex.Lock()
+		defer func() {
+			mpPodLock.mutex.Unlock()
+			releaseMPPodLock(mpPodUID)
+		}()
+
+		podCredentialsPath, err := pm.ensureCredentialsDirExists(podPath)
+		if err != nil {
+			klog.Errorf("Failed to create credentials directory for %q: %v", target, err)
+			return fmt.Errorf("Failed to create credentials directory for %q: %w", target, err)
+		}
+
+		credentialCtx.SetWriteAndEnvPath(podCredentialsPath, mppod.PathInsideMountpointPod(mppod.KnownPathCredentials))
+
+		_, _, err = pm.credProvider.Provide(ctx, credentialCtx)
```

Comment:
Since this procedure of refreshing credentials are done in two places, we should ideally extract it into a separate function